### ISSUE TITLE
Server status is 'active' when it's not ready - so can it be used?

### ISSUE THREAD

**Issue by planetf1** [odpi/egeria#5681]
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Current Behavior

A server reports as 'active' whilst it is still initializing

The notebook output shows as:
```
    Dev Platform is active
        Checking OMAG Server cocoMDSx
           ... cocoMDSx is configured
           ... cocoMDSx is not known - needs to be started
                Starting server cocoMDSx ...
```

Meantime a query to the platform for status shows as :
```
http --verify=no --pretty=format GET https://localhost:9445/open-metadata/platform-services/users/garygeeke/server-platform/servers/active        [16:16:32]
HTTP/1.1 200
Connection: keep-alive
Content-Type: application/json
Date: Fri, 10 Sep 2021 15:16:39 GMT
Keep-Alive: timeout=60
Transfer-Encoding: chunked

{
    "class": "ServerListResponse",
    "relatedHTTPCode": 200,
    "serverList": [
        "cocoMDSx"
    ]
}
```
cocoMDSx actually fails to connect to kafka after a timeout - with errors being reported both from the ongoing cohort connection & then Asset Catalog (probably the first OMAS). This concludes with:
```
Fri Sep 10 16:28:25 BST 2021 cocoMDSx Shutdown OMAG-ADMIN-0006 The cocoMDSx server has completed shutdown
```
At this point a query of active servers is empty as expected.

However in the interim (which can be 10 minutes+ given the kafka retry loop), the server is an intermediate state. Some of the services are active, but not all. For example OMRS is, OMASs are not ie:
```
$ http --verify=no --pretty=format GET https://localhost:9443/open-metadata/platform-services/users/garygeeke/server-platform/servers/active                          [17:01:08]
HTTP/1.1 200
Connection: keep-alive
Content-Type: application/json
Date: Fri, 10 Sep 2021 16:01:41 GMT
Keep-Alive: timeout=60
Transfer-Encoding: chunked

{
    "class": "ServerListResponse",
    "relatedHTTPCode": 200,
    "serverList": [
        "cocoMDS2",
        "cocoMDS5"
    ]
}

jonesn:~/ $ http --verify=no --pretty=format GET https://localhost:9443/open-metadata/platform-services/users/garygeeke/server-platform/servers/cocoMDS2/status                 [17:01:41]
HTTP/1.1 200
Connection: keep-alive
Content-Type: application/json
Date: Fri, 10 Sep 2021 16:01:56 GMT
Keep-Alive: timeout=60
Transfer-Encoding: chunked

{
    "active": true,
    "class": "ServerStatusResponse",
    "relatedHTTPCode": 200,
    "serverHistory": [
        {
            "endTime": "2021-09-10T13:27:55.680+00:00",
            "startTime": "2021-09-10T12:47:51.902+00:00"
        }
    ],
    "serverName": "cocoMDS2",
    "serverStartTime": "2021-09-10T15:46:18.722+00:00",
    "serverType": "Metadata Server"
}

jonesn:~/ $ http --verify=no --pretty=format GET https://localhost:9443/open-metadata/platform-services/users/garygeeke/server-platform/servers/cocoMDS2/services               [17:01:56]
HTTP/1.1 200
Connection: keep-alive
Content-Type: application/json
Date: Fri, 10 Sep 2021 16:02:18 GMT
Keep-Alive: timeout=60
Transfer-Encoding: chunked

{
    "class": "ServerServicesListResponse",
    "relatedHTTPCode": 200,
    "serverName": "cocoMDS2",
    "serverServicesList": [
        "Open Metadata Repository Services (OMRS)",
        "OMAG Server Operational Services"
    ]
}

```

Whilst this last query can dig into what services are active, it's not clear at a platform level when the intended initialization is complete, so when the platform is ready 

### Expected Behavior

A server should report as inactive or initializing whilst it is still getting ready.. and then update status to 'initialized' or similar so that we can decide at what point workload can be sent to the server.

For example in a environment such as k8s where we have multiple instances of a server platform, we want to be sure that the servers are ready before we allow workload (REST requests etc) to be sent their way

Ideally this would also be the case at the platform level where having servers set to autostart are part of the logical startup of the platform (for example this check would be easier in a k8s environment)

### Steps To Reproduce

* Stop kafka/zookeeper
* Run the coco pharma config notebook
* run the coco pharma notebook
* Observer results of API calls given above during this initializing state

### Environment

```markdown
Egeria: 3.2-SNAPSHOT (10 Sep)
```

**Comment by mandy-chessell** [comment]
This is an enhancement not a bug - the platform only knows that the server is active or not - it does not understand what is going on inside the server.

**Comment by planetf1** [comment]
Going back to what I think is needed from a user (caller perspective)
 a) is the server initialized and ready to take requests (for example we have returned successfully from the call to /instance)
 b) is the platform initialized and ready to take requests (for example we have started all the servers in our startup list, and they have completed as above) - this is particularly important in the k8s case where our modeling *currently* focusses on platform behaviour, not server
 
 For a) a few options might be
 * do not set state to active until the server is ready
 * add a third state 'activating' or initialization inbetween 
 * add an additional API call to return 'ready' status
 
 For b) we could
  * not do this change. Rather than relying on autostart, a 3rd party like the egeria operator needs to explicitly start each server. This will add more complexity but is possible. It would also mean some kind of modeling of the server itself, such as a server-specific service address so we can control request redirection to only 'ready' servers
  * Add a new API call to get 'ready' status
 
 In both cases we should clarify how this is done in the egeria docs, as I suspect similar will be needed in other orchestration environments.
